test/srv6-sighup-flow/src/main.rs

```diff
@@ -0,0 +1,172 @@
+#![feature(box_syntax)]
+#![feature(asm)]
+#[macro_use]
+extern crate log;
+extern crate simplelog;
+#[macro_use]
+extern crate lazy_static;
+#[macro_use]
+extern crate serde_derive;
+extern crate config;
+extern crate futures;
+extern crate netbricks;
+extern crate tokio;
+extern crate tokio_signal;
+use self::nf::*;
+use config::{Config, ConfigError, File, FileFormat};
+use log::Level;
+use netbricks::config::{basic_opts, read_matches, Atom};
+use netbricks::interface::*;
+use netbricks::operators::*;
+use netbricks::scheduler::*;
+use simplelog::{Config as SimpleConfig, LevelFilter, WriteLogger};
+use std::env;
+use std::fmt::Display;
+use std::fs::File as StdFile;
+use std::process;
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+use tokio::prelude::{Future, Stream};
+use tokio_signal::unix::{SIGUSR1, Signal, SIGHUP};
+mod nf;
+
+#[derive(Debug, Deserialize)]
+pub struct Foo {
+    pub baz: bool,
+    pub bar: Bar,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Bar {
+    pub baz: bool,
+}
+
+static DEFAULT_TOML: &'static str = r#"
+    baz = true
+    [bar]
+      baz = false
+"#;
+
+pub fn read_app_configuration() -> Result<Foo, ConfigError> {
+    let mut config = Config::new();
+    config.merge(File::from_str(DEFAULT_TOML, FileFormat::Toml))?;
+    config.try_into()
+}
+
+fn handle_signals(configuration: &'static Atom<Bar>) {
+    let sighup = Signal::new(SIGHUP).flatten_stream();
+    let sigusr1 = Signal::new(SIGUSR1).flatten_stream();
+
+    let on_signal = sighup.select(sigusr1).for_each(move |signal| {
+        match signal {
+            SIGHUP | SIGUSR1 => {
+                if let Ok(new_config) = read_app_configuration() {
+                    let mut nf_stuff = new_config.bar;
+                    nf_stuff.baz = true;
+                    configuration.set(nf_stuff);
+                    info!("SIGHUP HAPPNEDED WAT: {:?}", configuration.get());
```

Comment:
A nicer message on this `info!` is in order.

Reply:
Is it tho?